Sources/drawing.py

```diff
@@ -0,0 +1,418 @@
+        prov = ObservationLayer.dataProvider()
+        for obs in Observations:
+            for point in obs.Vertices:
+                feature = QgsFeature()
+                feature.setFields(self.fields[5])
+                layerPoint = QgsPointXY(point.X, point.Y)
+                feature.setGeometry(QgsGeometry.fromPointXY(layerPoint))
+                self.add_values(
+                    feature, obs, self.VX.LayerFieldsPovider.ObservationShapeFields)
+                points.append(feature)
+
+        prov.addFeatures(points)
+
+        ObservationLayer.loadNamedStyle(
+            self.plugin_dir + '\\Styles\\style_obs.qml')
+        ObservationLayer.updateExtents()
+        ObservationLayer.commitChanges()
+        QCoreApplication.processEvents()
+
+    def add_values(self, feature, entity, shapefields):
+        for attribute in feature.fields():
+            for value in shapefields:
+                attr = feature.fieldNameIndex(value.Key)
+                feature.setAttribute(attr, str(entity.GetValue(value)))
+                
+    def create_layers(self):
+        inspection_layer_name = self.wincan_layers[0]
+        section_layer_name = self.wincan_layers[1]
+        node_layer_name = self.wincan_layers[2]
+        node_nspection_layer_name = self.wincan_layers[3]
+        node_observation_layer_name = self.wincan_layers[4]
+        observation_layer_name = self.wincan_layers[5]
+        
+        date = datetime.now()
+        layers_path = self.VX.Project.Path + \
+            "\\Misc\\Exchange\\QGIS\\Layers" + date.strftime("%d_%m_%Y")
+        try:
+            os.makedirs(layers_path)
+        except FileExistsError:
+            pass
+
+        WKT = self.get_coordinate_system()
+        if WKT == "WKT:UNKNOWN":
+            EPSG = QgsCoordinateReferenceSystem()
+        else:
+            EPSG = QgsCoordinateReferenceSystem.fromEpsgId(int(WKT.split("\"")[-2]))
+        LFP = self.VX.LayerFieldsPovider
+        ShapeFields = [LFP.InspectionShapeFields, LFP.SectionShapeFields, LFP.NodeShapeFields,
+                       LFP.NodeInspectionShapeFields, LFP.NodeObservationShapeFields, LFP.ObservationShapeFields]
+        self.fields = []
+        nr = 0
+
+        for layer in self.wincan_layers:
+            layer_full_path = layers_path + "\\" + layer + ".shp"
+            self.fields.append(self.get_qgis_fields(ShapeFields[nr]))
+            if (layer == node_layer_name or layer == node_observation_layer_name or layer == observation_layer_name):
+                QgsVectorFileWriter(layer_full_path,
+                                    "UTF-8",
+                                    self.get_qgis_fields(ShapeFields[nr]),
+                                    QgsWkbTypes.Point,
+                                    EPSG,
+                                    "ESRI Shapefile")
+            if (layer == inspection_layer_name or layer == node_nspection_layer_name or layer == section_layer_name):
```

Comment:
almost the same value and action as above

Reply:
Changed